### PR TITLE
Feature/common error conditions  payment mismatch

### LINKED ISSUE #166: common-error-conditions - payment-mismatch - Expect a TotalPaymentDueMismatchError if payment does not match
**Feature**

Tests C1, C2 and B for common error conditions applicable to all implementations

**Use Case**

Run B with a totalPaymentDue that does not match the expected price (which will have been returned by C2) and expect a TotalPaymentDueMismatchError

In the Open Booking API spec: https://openactive.io/open-booking-api/EditorsDraft/#order-creation-orderquote-orderproposal-and-order-error-responses

### PR DESCRIPTION
**Pull request by civsiv**
closes #166

### CODE REVIEW

File: packages/openactive-integration-tests/test/helpers/feature-helper.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
-const config = require('config');
 const _ = require('lodash');
 
 const { Logger } = require('./logger');
 const { RequestState } = require('./request-state');
 const { FlowHelper } = require('./flow-helper');
 
-const BOOKABLE_OPPORTUNITY_TYPES_IN_SCOPE = global.BOOKABLE_OPPORTUNITY_TYPES_IN_SCOPE;
-const IMPLEMENTED_FEATURES = global.IMPLEMENTED_FEATURES;
+const { BOOKABLE_OPPORTUNITY_TYPES_IN_SCOPE, IMPLEMENTED_FEATURES } = global;
```
Comment (civsiv) [inline packages/openactive-integration-tests/test/helpers/feature-helper.js:7]: eslint suggestion and fix

File: packages/openactive-integration-tests/test/helpers/feature-helper.js [hunk 2]
```diff
@@ -1,12 +1,11 @@
-const config = require('config');
 const _ = require('lodash');
```
Comment (civsiv) [inline packages/openactive-integration-tests/test/helpers/feature-helper.js:1]: All of the changes below are made by the eslint autofix

File: packages/openactive-integration-tests/test/helpers/request-helper.js [hunk 3]
```diff
@@ -1,23 +1,22 @@
-const chakram = require("chakram");
```
Comment (civsiv) [inline packages/openactive-integration-tests/test/helpers/request-helper.js:1]: All of the changes below are made by the eslint autofix

File: packages/openactive-integration-tests/test/helpers/request-state.js [hunk 4]
```diff
@@ -1,33 +1,33 @@
-const RequestHelper = require("./request-helper");
```
Comment (civsiv) [inline packages/openactive-integration-tests/test/helpers/request-state.js:1]: All of the changes below are made by the eslint autofix

**Review by lukehesluke — APPROVED** [review]
Looks great @civsiv ! Just one thing - could you put "closes #{issue number}" in the description of this PR so that the correct issue gets closed when this is merged

File: packages/openactive-integration-tests/test/templates/b-req.js [hunk 5]
```diff
@@ -138,6 +138,23 @@ function createNoBrokerNameBReq(data) {
...
+ *
+ * @param {BReqTemplateData} data
+ */
+function createIncorrectTotalPaymentDuePriceBReq(data) {
+  const req = createStandardPaidBReq(data);
+  return {
+    ...req,
+    totalPaymentDue: {
+      ...req.totalPaymentDue,
+      price: req.totalPaymentDue.price - 1,
```
Comment (nickevansuk) [inline packages/openactive-integration-tests/test/templates/b-req.js:153]:
```suggestion
      price: req.totalPaymentDue.price + 1,
```

If we -1 we could end up with a negative price (as the price is in £ not pence), which is an interesting error state, but not one that we're testing for here?

**Review by nickevansuk — CHANGES_REQUESTED** [review]
One minor thing... otherwise great!

File: packages/openactive-integration-tests/test/templates/b-req.js (hunk 5 above)
Comment (civsiv) [reply packages/openactive-integration-tests/test/templates/b-req.js:153]: Yes! I had a dream about this, and was going to change it hah

**Pull request merged**